### PR TITLE
/api/v1/genes queries all genome builds by default

### PR DESCRIPTION
**Pull request by mhkc**
This PR fixes a bug where the user could not search for genes from grch38 in `/genes`

**How to test**:
1. goto `/genes`
2. search for "NOTCH2NLA" (in grch37 and 38)
3. search for "NOTCH2NLC" (in grch38)

**Expected outcome**:
- Genes should be found regardless of genome build
- Should only show one hit per gene, regardless if a gene is present in multiple genome builds

**Review:**
- [x] code approved by
- [x] tests executed by

### CODE REVIEW

**Comment by northwestwitch** [comment]
I've installed this PR on out stage server (Stockholm) so it's ready for testing also today, Otherwise I can test it and review tomorrow!

**Comment by northwestwitch** [comment]
@mhkc tried a quick search for NOTCH2NL and I get no suggestions for the one in build 38 only:
image

I'm pretty sure we should have genes in both builds there but I will check better tomorrow!

**Comment by mhkc** [comment]
The query is working but it is a bit wonky at times. I think it is the result of the initial query to find perfect matches. Note that NOTCH2NL is one of the aliases of NOTCH2NLA.

gene_query

Should I remove the initial perfect matches?

**Comment by dnil** [comment]
Sorry @northwestwitch - that is probably simply due to my aborted update test? It seems we don't have build:"38" on stage at the moment. Will fix!

**Comment by northwestwitch** [comment]
> Sorry @northwestwitch - that is probably simply due to my aborted update test? It seems we don't have build:"38" on stage at the moment. Will fix!

Yes I suspected it, immediately after writing this comment. Thanks!

**Comment by dnil** [comment]
Works after updating genes! Sorry about that!

Screenshot 2021-04-23 at 10 44 28

File: scout/adapter/mongo/hgnc.py [hunk 1]
```diff
@@ -116,20 +116,21 @@ def hgnc_genes(self, hgnc_symbol, build="37", search=False):
             pymongo.cursor
         """
         LOG.debug("Fetching genes with symbol %s" % hgnc_symbol)
+        build_query = {}
+        if str(build) in ["37", "38"]:
+            build_query["build"] = str(build)
```
Comment (dnil) [inline scout/adapter/mongo/hgnc.py:121]: :+1:

File: scout/server/blueprints/genes/controllers.py [hunk 2]
```diff
@@ -52,16 +52,16 @@ def gene(store, hgnc_id):
     return res
 
 
-def genes_to_json(store, query):
+def genes_to_json(store, query, build):
     """Fetch matching genes and convert to JSON."""
-    gene_query = store.hgnc_genes(query, search=True)
-    json_terms = [
-        {
+    gene_query = store.hgnc_genes(query, build, search=True)
```
Comment (dnil) [inline scout/server/blueprints/genes/controllers.py:57]: 👍

**Review by dnil — APPROVED** [review]
Works just fine! As you have identified a problem that could perhaps recur, a test would be super nice. That is no absolute requirement, just good practice!

File: scout/server/blueprints/genes/controllers.py [hunk 3]
```diff
@@ -52,16 +52,16 @@ def gene(store, hgnc_id):
...
+    json_terms = {
+        gene["hgnc_id"]: {
```
Comment (dnil) [inline scout/server/blueprints/genes/controllers.py:59]: The solutions appear very similar to me - was there some weird duplication going on, or was this just part of debugging? 😊

File: scout/server/blueprints/genes/controllers.py (hunk 2 above)
Comment (mhkc) [reply scout/server/blueprints/genes/controllers.py:57]: its basically passing on the build argument. I debugged the code at some time so there might appear to be more different than it actually is

File: scout/server/blueprints/genes/controllers.py (hunk 3 above)
Comment (mhkc) [reply scout/server/blueprints/genes/controllers.py:59]: its basically passing on the build argument. I debugged the code at some time so there might appear to be more different than it actually is

**Comment by mhkc** [comment]
@dnil I have added a few tests as well. Feel free to merge when you have the time

File: tests/adapter/mongo/test_gene_handler.py [hunk 4]
```diff
@@ -204,6 +204,58 @@ def test_get_genes_regex(real_adapter):
     assert sum(1 for i in res) == 3
 
 
+def test_get_genes_per_build(real_adapter):
```
Comment (dnil) [inline tests/adapter/mongo/test_gene_handler.py:207]: 👍

**Pull request merged**